### PR TITLE
Improve Section 508 accessibility compliance

### PR DESCRIPTION
**Pull request by fedde-s**
Implement compliance with Section 508 accessibility requirements on major pages by adding title and alt attributes and explicit relations to input field labels in various places.

Most of these fixes were part of PR #1219, ported to the hotfix branch as requested by @jjgao.

### CODE REVIEW

File: portal/src/main/webapp/WEB-INF/jsp/oncoprint/main.jsp [hunk 1]
```diff
@@ -96,13 +98,14 @@
...
                <div class="dropdown-menu" style="padding: 10px 5px; width: 270px;min-width: 270px;">
                    <form action="" style="margin-bottom: 0;">
-                       <span>Data type:</span><br>
-                       <input type="radio" name="datatype" style="margin-left:10px" value="sample"> Events per sample</input><br>
-                       <input type="radio" name="datatype" style="margin-left:10px" value="patient"> Events per patient</input><br>
-                       <br>
-                       <input type="checkbox" name="show_unaltered" value="show_unaltered"> Show unaltered columns</input><br>
-                       <input type="checkbox" name="show_whitespace" value="show_whitespace"> Show whitespace between columns</input><br>
-                       <input type="checkbox" name="show_clinical_legends" value="show_clinical_legends"> Show legends for clinical tracks</input>
+                       <fieldset style="border: 0; padding-bottom: 0;">
```
Comment (pieterlukasse) [inline portal/src/main/webapp/WEB-INF/jsp/oncoprint/main.jsp]: @fedde-s I'm assuming form message is unchanged. Can you test this?

File: portal/src/main/webapp/WEB-INF/jsp/plots_tab.jsp [hunk 2]
```diff
@@ -135,24 +135,24 @@
...
+                            <input type="radio" name="plots-y-data-type" value="clinical_attribute" title="y data type clinical attribute">Clinical Attribute
                         </div>
                         <div id="plots-y-spec"></div>
                     </div>
                     <div id="plots-sidebar-util-div" class="plots">
                         <h4>Utilities</h4>
-                        <h5>Search Case(s)</h5><input type="text" id="case_id_search_keyword" name="case_id_search_keyword" placeholder="Case ID.." onkeyup="search_case_id();"><br>
-                        <h5>Search Mutation(s)</h5><input type="text" id="mutation_search_keyword" name="mutation_search_keyword" placeholder="Protein Change.." onkeyup="search_mutation();"><br>
+                        <h5>Search Case(s)</h5><input type="text" id="case_id_search_keyword" name="case_id_search_keyword" placeholder="Case ID.." title="Search Case(s)" onkeyup="search_case_id();"><br>
+                        <h5>Search Mutation(s)</h5><input type="text" id="mutation_search_keyword" name="mutation_search_keyword" placeholder="Protein Change.." title="Search Case(s)" onkeyup="search_mutation();"><br>
```
Comment (pieterlukasse) [inline portal/src/main/webapp/WEB-INF/jsp/plots_tab.jsp]: title should be "Search Mutation(s)"

File: portal/src/main/webapp/js/src/dynamicQuery.js [hunk 3]
```diff
@@ -1448,17 +1448,17 @@ function addGenomicProfiles (genomic_profiles, targetAlterationType, targetClass
 
     if(targetClass == PROFILE_MRNA_EXPRESSION && downloadTab == false){
         var inputName = 'Z_SCORE_THRESHOLD';
-        profileHtml += "<div id='z_score_threshold' class='score_threshold'>Enter a z-score threshold &#177: "
+        profileHtml += "<div id='z_score_threshold' class='score_threshold' label='Enter a z-score threshold for mRNA expression'>Enter a z-score threshold &#177: "
         + "<input type='text' name='" + inputName + "' size='6' value='"
-                + window.zscore_threshold + "'>"
+                + window.zscore_threshold + "' title='z-score value'>"
```
Comment (pieterlukasse) [inline portal/src/main/webapp/js/src/dynamicQuery.js]: should be z-score threshold

File: portal/src/main/webapp/js/src/dynamicQuery.js [hunk 4]
```diff
@@ -1448,17 +1448,17 @@ function addGenomicProfiles (genomic_profiles, targetAlterationType, targetClass
...
         + "</div>";
     }
 
     if(targetClass == PROFILE_PROTEIN_EXPRESSION && downloadTab == false){
         var inputName = 'RPPA_SCORE_THRESHOLD';
-        profileHtml += "<div id='rppa_score_threshold' class='score_threshold'>Enter a RPPA z-score threshold &#177: "
+        profileHtml += "<div id='rppa_score_threshold' class='score_threshold' label='Enter a z-score threshold for RPPA expression'>Enter a RPPA z-score threshold &#177: "
         + "<input type='text' name='" + inputName + "' size='6' value='"
-                + window.rppa_score_threshold + "'>"
+                + window.rppa_score_threshold + "' title='z-score value'>"
```
Comment (pieterlukasse) [inline portal/src/main/webapp/js/src/dynamicQuery.js]: should be z-score threshold

**Comment by pieterlukasse** [comment]
@fedde-s : I added some minor comments. Please check.

File: portal/src/main/webapp/WEB-INF/jsp/oncoprint/main.jsp (hunk 1 above)
Comment (fedde-s) [inline portal/src/main/webapp/WEB-INF/jsp/oncoprint/main.jsp]: There is no HTTP action for this form, the selection is interpreted by JavaScript:
```JavaScript
	    // initialize radio buttons
	    $(toolbar_selector).find('#oncoprint_diagram_view_menu')
				   .find('input[type="radio"][name="datatype"][value="'+
				   (State.using_sample_data ? "sample" : "patient") + '"]')
				   .prop("checked", true);
	    
	    $(toolbar_selector).find('#oncoprint_diagram_view_menu')
				   .find('input[type="radio"][name="datatype"]').change(function(e) {
		State.using_sample_data = $(toolbar_selector).find('#oncoprint_diagram_view_menu')
				   .find('input[type="radio"][name="datatype"]:checked').val() === 'sample';
		if (State.using_sample_data) {
		    State.setDataType('sample');
		} else {
		    State.setDataType('patient');
		}
		updateHeaderBtnText();
		updateDownloadIdOrderText();
	    });
```

Comment (pieterlukasse) [inline portal/src/main/webapp/WEB-INF/jsp/oncoprint/main.jsp]: ok.  Should work, but good to test anyway since the html structure changed.

Comment (fedde-s) [inline portal/src/main/webapp/WEB-INF/jsp/oncoprint/main.jsp]: Seems to work fine here!

**Comment by fedde-s** [comment]
@jjgao Could you look if you find this ready to merge too?

**Comment by jjgao** [comment]
@fedde-s could you update your branch?

**Comment by fedde-s** [comment]
@jjgao: done.

**Comment by fedde-s** [comment]
Ah, no, since commit aee9eb0c3ad9722e6ff450de53dd9d25f7427403 they require the stable id of fusion data to be ‘fusions’. So this documentation change introduces a bug.

**Pull request merged**